fe/fe-core/src/main/java/org/apache/iceberg/StarRocksIcebergTableScan.java

```diff
@@ -321,6 +323,45 @@ private FileScanTask toFileScanTask(DataFile dataFile) {
                 residuals);
     }
 
+    private boolean shouldPlanLocally(List<ManifestFile> manifests, boolean loadColumnStats) {
+        return (planMode == PlanMode.AUTO && loadColumnStats) || shouldPlanLocally(manifests);
+    }
+
+    // TODO(stephen): add more strategies
+    private boolean shouldPlanLocally(List<ManifestFile> manifests) {
+        switch (planMode) {
+            case LOCAL:
+                return true;
+
+            case DISTRIBUTED:
+                return manifests.isEmpty();
```

Comment:
just return false?

Reply:
if there are no manifest, we should use local mode and short circuit